ts/private/ts_project.bzl

```diff
@@ -6,31 +6,25 @@ load("@aspect_bazel_lib//lib:paths.bzl", "to_output_relative_path")
 load("@aspect_bazel_lib//lib:platform_utils.bzl", "platform_utils")
 load("@aspect_rules_js//js:libs.bzl", "js_lib_helpers")
 load("@aspect_rules_js//js:providers.bzl", "JsInfo", "js_info")
-load("@aspect_rules_js//npm:providers.bzl", "NpmPackageStoreInfo")
 load("@bazel_skylib//lib:dicts.bzl", "dicts")
 load(":options.bzl", "OptionsInfo", "transpiler_selection_required")
 load(":ts_config.bzl", "TsConfigInfo")
 load(":ts_lib.bzl", "COMPILER_OPTION_ATTRS", "OUTPUT_ATTRS", "STD_ATTRS", _lib = "lib")
 load(":ts_validate_options.bzl", _validate_lib = "lib")
 
 # Forked from js_lib_helpers.js_lib_helpers.gather_files_from_js_providers to not
-# include any sources; only transitive declarations & npm linked packages
-def _gather_declarations_from_js_providers(targets):
+# include any sources; only transitive types & npm sources
+def _gather_types_from_js_infos(targets):
     files_depsets = []
     files_depsets.extend([
-        target[JsInfo].transitive_declarations
+        target[JsInfo].transitive_types
         for target in targets
-        if JsInfo in target and hasattr(target[JsInfo], "transitive_declarations")
+        if JsInfo in target and hasattr(target[JsInfo], "transitive_types")
     ])
     files_depsets.extend([
-        target[JsInfo].transitive_npm_linked_package_files
+        target[JsInfo].npm_sources
         for target in targets
-        if JsInfo in target and hasattr(target[JsInfo], "transitive_npm_linked_package_files")
-    ])
-    files_depsets.extend([
-        target[NpmPackageStoreInfo].transitive_files
-        for target in targets
-        if NpmPackageStoreInfo in target and hasattr(target[NpmPackageStoreInfo], "transitive_files")
+        if JsInfo in target and hasattr(target[JsInfo], "npm_sources")
```

Comment:
Why do we need the `hasattr`? Doesn't the factory method always set it?

Reply:
It does but we can't say for certain that other custom rule sets that derive from rules_js will set it so we check as a defence-in-depth.